### PR TITLE
Close #203: Software update: add an icon for advanced settings, which can call a pop-up window with additional information

### LINKED ISSUE #203: Software update: add an icon for advanced settings, which can call a pop-up window with additional information
It is necessary to describe which binaries are required to install the update.

### PR DESCRIPTION
**Pull request by TheSomeMan**
@laurijamsa could you please check how the new icon/tooltip/popup looks like?

image

image

image

### CODE REVIEW

**Comment by laurijamsa** [comment]
The outlook requires some tuning. Sini-Maaria can create guidelines on how it should look like.

**Comment by sini-maaria** [comment]
Please, use the same "i" icon style as on Ruuvi Station web app. (Let me know if you need a file for that.)

You popup is fine, if it is default style for the Gateway Setup. Just align the text content to the left (instead of justify) and add some space to the top before the title.

Screenshot 2022-06-20 at 8 38 49
Screenshot 2022-06-20 at 8 38 54

**Comment by TheSomeMan** [comment]
@sini-maaria thanks, 
please attach the file for "I" icon

**Comment by sini-maaria** [comment]
We will discuss with Lauri about this UI and will get back to you.

**Comment by ojousima** [comment]
Please re-request review from me once this is ready

**Comment by TheSomeMan** [comment]
image

image

image

**Comment by TheSomeMan** [comment]
image

image

**Comment by TheSomeMan** [comment]
image

image

File: src/index.html [hunk 1]
```diff
@@ -478,18 +478,47 @@ <h2>
...
+			<div id="popup-info-icon-software_update" class="info-icon-overlay">
+				<div class="info-icon-popup">
+					<h1>How to install custom firmware</h1>
+					<a class="info-icon-popup-close" onclick="window.history.back()">&times;</a>
+					<div class="info-icon-content">
+						<p>
+							<span lang="en">You need to prepare firmware binaries (ruuvi_gateway_esp.bin,
+								fatfs_gwui.bin and fatfs_nrf52.bin) and put them on your HTTP server.
+							</span>
+							<span lang="fi">Sinun on valmisteltava laiteohjelmiston binaarit
```
Comment (ojousima) [inline src/index.html:505]: `Sinun on valmistettava laiteohjelmiston binäärit (ruuvi_gateway_esp.bin, fatfs_gwui.bin ja fatfs_nrf52.bin) ja ladata ne HTTP-palvelimellesi.`

File: src/index.html [hunk 2]
```diff
@@ -663,84 +692,59 @@ <h2>
 				</div>
 				<div>&nbsp;</div>
 				<div id="remote_cfg-options" class="indent hidden">
-					<label for="remote_cfg-base_url">
-						<span lang="en">Specify URL of json file with the configuration or URL of the folder with the set of json files for each Gateway:</span>
-						<span lang="fi">Määritä json-tiedoston URL-osoite kokoonpanon kanssa tai sen kansion URL-osoite, jossa on json-tiedostot jokaiselle yhdyskäytävälle:</span>
+					<label for="remote_cfg-base_url" class="info-icon-container">
+						<div class="info-icon-left-column">
+							<span lang="en">Specify URL of json file with the configuration or URL of the folder with the set of json files for each Gateway:</span>
+							<span lang="fi">Määritä json-tiedoston URL-osoite kokoonpanon kanssa tai sen kansion URL-osoite, jossa on json-tiedostot jokaiselle yhdyskäytävälle:</span>
```
Comment (ojousima) [inline src/index.html:698]: `Määritä json-tiedoston URL-osoite kokoonpanon kanssa tai sen kansion URL-osoite, jossa on json-tiedostot jokaiselle Gatewaylle:`

"yhdyskäytävä" would be a good translation, but we don't translate "Ruuvi Gateway" product name

File: src/index.html [hunk 3]
```diff
@@ -764,6 +768,39 @@ <h2>
...
+					<div class="info-icon-content">
+						<p>
+							<span lang="en">
+								If you specify a folder name, it will first attempt to read the configuration file
+								for this Gateway from &lt;GW_MAC&gt;.json (for example, AABBCCDDEEFF.json).
+								If that file fails, an attempt will be made to read the overall configuration
+								from gw_cfg.json.
+							</span>
+							<span lang="fi">
+								Jos määrität kansion nimen, se yrittää ensin lukea tämän yhdyskäytävän
```
Comment (ojousima) [inline src/index.html:783]: `Jos määrität kansion nimen, se yrittää ensin lukea tämän Gatewayn määritystiedoston &lt;GW_MAC&gt;.json-tiedostosta (esimerkiksi AABBCCDDEEFF.json). Jos tiedosto epäonnistuu, koko kokoonpano yritetään lukea tiedostosta gw_cfg.json.`

**Review by ojousima — CHANGES_REQUESTED** [review]
Thanks, please see changed translations

**Pull request merged**